### PR TITLE
enable propagation warnings before postgres vanilla tests

### PR DESCRIPTION
**Pull request by aykut-bozkurt**
DESCRIPTION: Adds the GUC enable_unsupported_feature_messages to control some of the citus related messages.

This PR adds the GUC to stop some of the citus messages that causes some of the postgres tests to fail. It is a subtask for the main PR #6018. It is intended to be set false only before postgres vanilla tests to prevent some of the tests from failing.

### CODE REVIEW

File: src/backend/distributed/commands/function.c [hunk 1]
```diff
@@ -1387,7 +1387,11 @@ PostprocessCreateFunctionStmt(Node *node, const char *queryString)
 
 	if (errMsg != NULL)
 	{
-		RaiseDeferredError(errMsg, WARNING);
+		if (EnablePropagationWarnings)
```
Comment (onderkalaci) [inline src/backend/distributed/commands/function.c:1390]: maybe `EnableWarningPropagation` is a better name?

File: src/backend/distributed/commands/table.c [hunk 2]
```diff
@@ -439,9 +439,18 @@ PreprocessAlterTableStmtAttachPartition(AlterTableStmt *alterTableStatement,
...
-			bool partitionMissingOk = false;
+
+			/*
+			 * Citus should not throw error for non-existing objects, let Postgres do that.
+			 * Otherwise, Citus might throw a different error than Postgres, which we don't want.
+			 */
+			bool partitionMissingOk = true;
 			Oid partitionRelationId = RangeVarGetRelid(partitionCommand->name, lockmode,
 													   partitionMissingOk);
+			if (!OidIsValid(partitionRelationId))
```
Comment (onderkalaci) [inline src/backend/distributed/commands/table.c:450]: is this change really relevant to this PR? Maybe move back this part to the main PR for now?

File: src/backend/distributed/commands/type.c [hunk 3]
```diff
@@ -352,10 +352,14 @@ AlterTypeStmtObjectAddress(Node *node, bool missing_ok)
 	AlterTableStmt *stmt = castNode(AlterTableStmt, node);
 	Assert(AlterTableStmtObjType_compat(stmt) == OBJECT_TYPE);
 
+	ObjectAddress *address = palloc0(sizeof(ObjectAddress));
```
Comment (onderkalaci) [inline src/backend/distributed/commands/type.c:355]: same here: Does this change really belong to this PR? Maybe save these changes to other PR. And, in this PR mostly focus on the new GUC

File: src/backend/distributed/shared_library_init.c [hunk 4]
```diff
@@ -1069,6 +1070,17 @@ RegisterCitusConfigVariables(void)
...
 
+	DefineCustomBoolVariable(
+		"citus.enable_propagation_warnings",
+		gettext_noop("Enables citus propagation warnings. It is intended to "
+					 "be used before vanilla tests to stop propagation warnings."),
+		NULL,
+		&EnablePropagationWarnings,
+		true,
+		PGC_USERSET,
+		GUC_SUPERUSER_ONLY | GUC_NO_SHOW_ALL,
```
Comment (onderkalaci) [inline src/backend/distributed/shared_library_init.c:1081]: I probably wouldn't mind if a regular user can control this behavior. So, probably not  GUC_SUPERUSER  required?

**Review by onderkalaci — COMMENTED** [review]
I'd rather focus on pieces one-by-one. So, in this PR lets only deal with the changes that are relevant to the GUC. And, others, we can move to another PR.

For this PR, what about the warnings on `DeferErrorIfHasUnsupportedDependency`? Specifically talking, the changes on `PostprocessCreateDistributedObjectFromCatalogStmt ` on #6018?

**Comment by onderkalaci** [comment]
> DESCRIPTION: We disable propagation warnings before postgres vanilla tests to prevent citus related propagation warnings.

Descriptions should be at max 78 chars of length, and in the form of entries here: https://github.com/citusdata/citus/blob/main/CHANGELOG.md

So, for example, this one might be: `Adds citus.... to control unsupported feature messages for non-distributed objects `

(or something along these lines)

File: src/backend/distributed/commands/table.c (hunk 2 above)
Comment (aykut-bozkurt) [reply src/backend/distributed/commands/table.c:450]: I ll move to another pr.

File: src/backend/distributed/metadata/metadata_sync.c [hunk 5]
```diff
@@ -2254,9 +2254,17 @@ GenerateGrantOnFunctionQueriesFromAclItem(Oid functionOid, AclItem *aclItem)
 		}
 		else
 		{
-			ereport(ERROR, (errmsg("unsupported prokind"),
-							errdetail("GRANT commands on procedures are propagated only "
-									  "for procedures and functions.")));
+			if (EnableWarningsPropagation)
+			{
+				ereport(ERROR, (errmsg("unsupported prokind"),
```
Comment (onderkalaci) [inline src/backend/distributed/metadata/metadata_sync.c:2259]: I think we should NOT guard any ERRORs with this GUC.

Can we fix this error without the GUC?

File: src/backend/distributed/shared_library_init.c [hunk 6]
```diff
@@ -1069,6 +1070,17 @@ RegisterCitusConfigVariables(void)
 		GUC_NO_SHOW_ALL,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.enable_warnings_propagation",
```
Comment (onderkalaci) [inline src/backend/distributed/shared_library_init.c:1074]: This name is too generic.

maybe `citus.enable_unsupported_feature_messages` or such?

File: src/backend/distributed/metadata/metadata_sync.c (hunk 5 above)
Comment (aykut-bozkurt) [reply src/backend/distributed/metadata/metadata_sync.c:2259]: I agree, we should not guard the error to not change citus behaviour, so I removed that change from this PR. I will solve it on future sub PRs.
